src/robot/can_adapter/adapter_test.py

```diff
@@ -0,0 +1,36 @@
+"""
+Test module for the CAN adapter.
+"""
+import asyncio
+from unittest.mock import MagicMock
+
+import can
+import pytest
+
+from src.robot.can_adapter.adapter import CANAdapter
+
+
+@pytest.mark.asyncio
+async def test_can_adapter():
+    """
+    Make sure that the CAN adapter can send and receive messages.
+    """
+    bus = can.Bus('vcan0', bustype='virtual', receive_own_messages=True)
+    adapter = CANAdapter(can_bus=bus)
+
+    handler1 = MagicMock()
+    handler2 = MagicMock()
+
+    adapter.register_handlers({
+        42: handler1,
+        10: handler2,
+    })
+    adapter.send(42, b'my data')
+    adapter.send(10, b'my other data')
```

Comment:
How does python-can function with a payload bigger than 8 bytes, in normal, non CAN FD (0-64 bytes) frames?
And how does it manage CAN FD frames with more than 64 bytes of payload?

Reply:
This test case was actually wrong, I modified it so it only test with payloads with a size < 8 bytes.

The check=true will make python-can throw an exception for msg sent with a payload that have more than 8 bytes for non can fd frames.x